### PR TITLE
Typescript migration

### LINKED ISSUE #6: Migrate code to TypeScript
Migrating code to TypeScript will make the contribution easier for the project. 

It will also enable us to deliver a compiled code which will help project to be used with earlier versions of node. Due to ES2015 features used in the code, node engine is restricted for versions <8.3.0

### PR DESCRIPTION
**Pull request by onderceylan**
fixes #6

### CODE REVIEW

File: src/helpers/file.ts [hunk 1]
```diff
@@ -75,8 +96,8 @@ const pathExists = (filePath, mode) => {
   });
 };
 
-const makeDir = folderPath => {
-  return new Promise((resolve, reject) => {
+const makeDir = (folderPath: string): Promise<string> => {
+  return new Promise((resolve: Function, reject: Function): void => {
```
[the remark this answers is not in the document]
Comment (sneas) [reply src/helpers/file.ts]: It's better to

```ts
return new Promise<string>((resolve, reject) => {...});
```

This way TS will automatically validate resolve and reject calls

File: src/cli.ts [hunk 2]
```diff
@@ -53,14 +52,18 @@ $ pwa-asset-generator --help
...
+    // Default vars can be number and boolean too but declared to be string in type defs
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
+    flags: constants.FLAGS,
   },
 );
 const logger = preLogger('cli', cli.flags);
 
-(async () => {
+(async (): Promise<void> => {
```
Comment (sneas) [inline src/cli.ts]: It is not necessary to add return types for anonymous functions.

Comment (onderceylan) [reply src/cli.ts]: Project uses @typescript-eslint eslint settings which recommends explicit-function-return-type and warns when it's not there. That's the main reason of return types everywhere.

File: src/helpers/file.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (onderceylan) [reply src/helpers/file.ts]: Yes, that's right. I will remove Function types but I prefer to keep consistent return types for all functions for the reason explained here: https://github.com/onderceylan/pwa-asset-generator/pull/44#discussion_r327614990.

File: src/helpers/flags.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    return '.';
+  }
+  return output;
+};
+
+const getDefaultOptions = (): Options => {
+  const flags = constants.FLAGS as Record<
+    keyof Options,
+    { type: string; alias: string; default?: string | number | boolean }
+  >;
```
Comment (sneas) [inline src/helpers/flags.ts]: Hardcore typescript ⚡️:)

**Comment by onderceylan** [comment]
Thanks @sneas @ofcyln!

**Pull request merged**